server/src/lib/vnc.rs

```diff
@@ -65,12 +83,30 @@ impl PropolisVncServer {
         }
     }
 
-    pub async fn update(&self, config: &Config, is_valid: bool) {
+    pub async fn update(
+        &self,
+        vnc_server: &VncServer<PropolisVncServer>,
+        config: &Config,
+        is_valid: bool,
+    ) {
         if is_valid {
             debug!(self.log, "updating framebuffer");
-            let (addr, w, h) = config.get_framebuffer_info();
-            let fb = RamFb::new(addr, w, h);
-            self.initialize_framebuffer(fb).await;
+            let fb_spec = config.get_framebuffer_spec();
+            let fb = RamFb::new(fb_spec);
+            self.initialize_framebuffer(fb.clone()).await;
+
+            match fourcc::fourcc_to_pixel_format(fb.fourcc) {
+                Ok(pf) => {
+                    vnc_server.set_pixel_format(pf).await;
```

Comment:
Small nit: Does this routine require that the caller have exclusive access to `vnc_server`? By itself and without that requirement, it looks like there's a race here (if two callers supply different configs, the last one to call `initialize_framebuffer` won't necessarily be the last one to call `set_pixel_format`, so the two could get out of sync).

The only caller I can see locks the `vnc_server` it provides before calling this function, so I don't think there's a bug here so much as a small seam that could tear later. 

Would it make sense to make `vnc_server` a `&mut` reference to clarify that only one caller at a time can call this with the same server?